miio/airdehumidifier.py

```diff
@@ -185,7 +185,7 @@ def status(self) -> AirDehumidifierStatus:
         values = self.get_properties(properties, max_properties=1)
 
         return AirDehumidifierStatus(
-            defaultdict(lambda: None, zip(properties, values)), self.info()
+            defaultdict(lambda: None, zip(properties, values)), self._fetch_info()
```

Comment:
This is wrong, the `_fetch_info()` is device-class internals and should not be accessed by any other class.

Reply:
Well how do you then suggest to prevent the loop: status --> info --> _initialize_descriptors--> status --> etc.
I can just remove the info call, but I am not sure what will then break downstream.

Reply:
I'm not sure what's the best approach right off the bat (especially as genericmiot constructs status containers dynamically), but for statically defined containers it would be enough to access the status container class without really initializing one, right?

```
In [5]: from miio.integrations.vacuum.roborock import VacuumStatus

In [6]: vars(VacuumStatus)
Out[6]: 
mappingproxy({'__module__': 'miio.integrations.vacuum.roborock.vacuumcontainers',
<snip>
              '_sensors': {'state_code': SensorDescriptor(id='VacuumStatus.state_code', type=<class 'int'>, name='State code', property='state_code', unit=None),
               'state': SensorDescriptor(id='VacuumStatus.state', type=<class 'str'>, name='State', property='state', unit=None),
<snip>
              '_settings': {'fanspeed': NumberSettingDescriptor(id='VacuumStatus.fanspeed', name='Fanspeed', property='fanspeed', unit='%', min_value=0, max_value=100, step=1, range_attribute=None, type=<SettingType.Number: 2>),
<snip>
```

Probably the simplest way to access the status container would be using the return type annotation like this:
```
_status_container = self_.status.__annotations__["return"]
```

Do you think that could work for the time being until we figure out a better solution for that?

Reply:
I just tried it, but that does not include the embeded sensors:

```
from miio import RoborockVacuum

r = RoborockVacuum(ip = "192.168.1.IP", token = "TokenTokenToken")
status = r.status.__annotations__["return"]
print(status.sensors(status))
```

That is missing the embeded sensors/settings....

Reply:
Ouch, that's indeed annoying... The real solution would be doing the "embedding" of status container _classes_ outside of the status query (maybe inside `__init__`), and performing the update on the container prior to returning it in `status()`. This way:
1. we can introspect the available information without performing any I/O.
2. if some of the method calls required to embedded containers fail, we can remove those queries from future update cycles.

The question is how/where to do this sanely, I don't have an answer to that right now.

Reply:
I do see what you mean, and think it is a good path forward,
but I think at this point it gets a bit to complicated for me.

I think it would be more efficient if you would implement those changes.
Than we also do not need to have long wait times in between us working on it.

You could do one of the following:
- make the changes and push them to this branch.
- copy this code as a start, make your own branch with a new PR and close this one
- First merge this as a starting point and then make the changes later to get rid of the _fetch_info calls in device classes (can be added as a TO DO)